nixos/modules/config/i18n.nix

```diff
@@ -40,6 +40,22 @@ with lib;
         '';
       };
 
+      defaultCharacterSet = mkOption {
+        type = types.str;
+        default = lib.last (builtins.split "\\." (
+          builtins.replaceStrings [ "utf8" "utf-8" "UTF8" ] [ "UTF-8" "UTF-8" "UTF-8" ] config.i18n.defaultLocale
+        ));
+        defaultText = literalExpression ''
+          lib.last (builtins.split "\\." (
+            builtins.replaceStrings [ "utf8" "utf-8" "UTF8" ] [ "UTF-8" "UTF-8" "UTF-8" ] config.i18n.defaultLocale
+          ))
+        '';
+        example = "UTF-8";
+        description = lib.mdDoc ''
+          The default character set.
```

Comment:
This description is a bit short. I am not sure when I would need to change this to something else.

Reply:
Yes, exactly.

Reply:
The description is now set to

```markdown
The default character set. Only set this if {option}`i18n.defaultLocale` does not contain a character set.
```